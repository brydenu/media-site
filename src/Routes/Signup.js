import React, { useState, useContext } from "react";
import { useNavigate } from "react-router-dom";
import AppContext from "../Helpers/Context";
import Card from "../Common/Card";
import Form from "../Common/Form";
import Backend from "../api";

export default function Signup() {
    const { setToken } = useContext(AppContext).tokenState;
    const { setUser } = useContext(AppContext).userState;
    const navigate = useNavigate();
    const [userInput, setUserInput] = useState({
        username: "",
        firstName: "",
        lastName: "",
        password: "",
    });
    const headerContent = <h1 className="signup-title">Sign up</h1>;
    const formFields = [
        {
            label: "Username",
            type: "text",
            fieldName: "username",
            required: true,
        },
        {
            label: "First Name",
            type: "text",
            fieldName: "firstName",
            required: false,
        },
        {
            label: "Last Name",
            type: "text",
            fieldName: "lastName",
            required: false,
        },
        {
            label: "Password",
            type: "password",
            fieldName: "password",
            required: true,
        },
    ];
    const handleSubmit = async (evt) => {
        evt.preventDefault();
        const newToken = await Backend.registerUser(userInput);
        setToken(newToken);
        setUser(userInput);
        navigate("/");
    };
    const bodyContent = (
        <Form
            fields={formFields}
            formClass="signup"
            handleSubmit={handleSubmit}
            inputState={[userInput, setUserInput]}
            buttonLabel="Register"
        />
    );
    return (
        <div className="main-signup">
            <Card
                header={headerContent}
                body={bodyContent}
                cardClass="signup"
            />
        </div>
    );
}
