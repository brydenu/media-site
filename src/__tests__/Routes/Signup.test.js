import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Signup from "../../Routes/Signup";
import TestContext from "../../contextForTests";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <TestContext>
                <Signup />
            </TestContext>
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <TestContext>
                <Signup />
            </TestContext>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});
