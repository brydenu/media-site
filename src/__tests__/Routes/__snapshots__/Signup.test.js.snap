// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-signup"
  >
    <div
      class="card-wrapper signup"
    >
      <div
        class="card-content signup"
      >
        <div
          class="card-header "
        >
          <h1
            class="signup-title"
          >
            Sign up
          </h1>
        </div>
        <div
          class="card-body signup"
        >
          <form
            class="custom-form signup"
          >
            <div
              class="form-group username"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                autocomplete="off"
                name="username"
                placeholder=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group firstName"
            >
              <label
                for="firstName"
              >
                First Name
              </label>
              <input
                autocomplete="off"
                name="firstName"
                placeholder=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group lastName"
            >
              <label
                for="lastName"
              >
                Last Name
              </label>
              <input
                autocomplete="off"
                name="lastName"
                placeholder=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group password"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="off"
                name="password"
                placeholder=""
                type="password"
                value=""
              />
            </div>
            <div
              class="clickables-group"
            >
              <div
                class="checkbox-group"
              >
                <div
                  class="checkbox-outer"
                >
                  <div
                    class="custom-checkbox remembered-false"
                  />
                </div>
                <p
                  class="remembered-label"
                >
                  Stay signed in on this browser?
                </p>
              </div>
              <div
                class="form-error hiding-error"
              />
              <button
                class="form-button"
                type="submit"
              >
                Register
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
